aten/src/ATen/native/TensorAdvancedIndexing.cpp

```diff
@@ -72,6 +73,54 @@
 #include <vector>
 
 namespace at {
+namespace native {
+
+inline std::string shapes_as_str(TensorList tensors) {
```

Comment:
for reference, in GitHub's review UI moving functions around like this makes reviewer's life harder as we have to manually verify that there were no substantive changes

Reply:
Should this still be taking a TensorList?

Reply:
> Is it better just to add the function signature there?

Well, there are a few ways you can do it. One is to forward declare. Another is to do the code motion in a separate PR stacked before this one.